pkgs/by-name/go/goldboot/package.nix

```
@@ -0,0 +1,57 @@
+{
+  fetchFromGitHub,
+  rustPlatform,
+  lib,
+  versionCheckHook,
+  pkg-config,
+  zstd,
+  OVMF,
+  qemu,
+  qemu-utils,
+  openssl,
+  nix-update-script,
+}:
+
+rustPlatform.buildRustPackage (finalAttrs: {
+  pname = "goldboot";
+  version = "0.0.10";
+
+  src = fetchFromGitHub {
+    owner = "fossable";
+    repo = "goldboot";
+    rev = "goldboot-v${finalAttrs.version}";
+    hash = "sha256-O9yhyJZpjQxC0HP43RsOgPMOKp6d23SNhMLiGtmwXzs=";
+  };
+
+  useFetchCargoVendor = true;
+  cargoHash = "sha256-NF0Fj+r6qWcM4VEIm1fzveZuz6MIaG32Z+zBfSMC/t4=";
+
+  buildAndTestSubdir = "goldboot";
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [
+    zstd
+    OVMF
+    qemu
+    qemu-utils
+    openssl
+  ];
+
+  # Tests require networking, so skip them for now
+  doCheck = false;
+
+  doInstallCheck = true;
+  nativeInstallCheck = [ versionCheckHook ];
```

```suggestion
  nativeInstallCheckInputs = [ versionCheckHook ];
```
At the moment it would just set the `nativeInstallCheck` env variable

---

oops sorry abt bad suggestion